src/Generator/Generators/CSharp/CSharpMarshal.cs

```diff
@@ -580,13 +580,16 @@ public override bool VisitPointerType(PointerType pointer, TypeQualifiers quals)
                 if (isRefParam)
                 {
                     var typeName = Type.TypePrinterDelegate(finalPointee);
-
-                    if (param.IsInOut)
-                        Context.Before.WriteLine("{0} _{1} = {1};", typeName, param.Name);
+                    if (Context.Function.OperatorKind != CXXOperatorKind.Subscript)
```

Comment:
Positive conditions are a little faster to understand when reading code so invert the conditional.